File: email-backup.java/src/test/java/demo/SessionAuthenticatorTest.java
package demo;

import org.testng.annotations.Test;

import javax.mail.PasswordAuthentication;

import static org.testng.Assert.assertEquals;

public class SessionAuthenticatorTest {

    private static final String EMAIL = "email";
    private static final String PASSWORD = "password";

    @Test
    public void constructorTest() {
        SessionAuthenticator authenticator = new SessionAuthenticator(EMAIL, PASSWORD);

        PasswordAuthentication authentication = authenticator.getPasswordAuthentication();

        assertEquals(authentication.getUserName(), EMAIL);
        assertEquals(authentication.getPassword(), PASSWORD);
    }
}

File: email-backup.kotlin/src/test/kotlin/demo/SessionAuthenticatorTest.kt
package demo

import org.testng.Assert.assertEquals
import org.testng.annotations.Test

public class SessionAuthenticatorTest {

    private val EMAIL = "email"
    private val PASSWORD = "password"

    @Test
    public fun constructorTest() {
        val authenticator = SessionAuthenticator(EMAIL, PASSWORD)

        val authentication = authenticator.getPasswordAuthentication()

        assertEquals(authentication.userName, EMAIL)
        assertEquals(authentication.password, PASSWORD)
    }
}
